src/nova/galaxy/parameters.py

```diff
@@ -18,3 +20,83 @@ def change_input_value(self, name: str, new_value: Any) -> None:
 
     def remove_input(self, name: str) -> None:
         self.inputs.pop(name)
+
+
+class WorkflowParameters:
+    """Handles workflow parameters using explicit bioblend-style approach."""
+
+    def __init__(self) -> None:
+        self.workflow_inputs: Dict[str, Any] = {}
+        self.step_params: Dict[str, Dict[str, Any]] = {}
+
+    def add_workflow_input(self, input_id: str, value: Any) -> None:
+        """Add a workflow-level input.
+
+        Parameters
+        ----------
+        input_id : str
+            The workflow input ID (e.g., "0", "1")
+        value : Any
+            The input value (Dataset, DatasetCollection, or simple value)
+        """
+        if isinstance(value, Dataset):
+            if not value.id:
+                raise ValueError(f"Dataset for workflow input '{input_id}' must have an ID")
+            self.workflow_inputs[input_id] = {"src": "hda", "id": value.id}
+        elif isinstance(value, DatasetCollection):
+            if not value.id:
+                raise ValueError(f"DatasetCollection for workflow input '{input_id}' must have an ID")
+            self.workflow_inputs[input_id] = {"src": "hdca", "id": value.id}
+        else:
+            # Simple values (strings, booleans, etc.)
+            self.workflow_inputs[input_id] = value
+
+    def add_step_param(self, step_id: str, param_path: str, value: Any) -> None:
+        """Add a step-level parameter.
+
+        Parameters
+        ----------
+        step_id : str
+            The workflow step ID (e.g., "2", "4")
+        param_path : str
+            The parameter path within the step (e.g., "input", "series_0|input_mode|export_folder")
```

Comment:
Not an issue, just a comment. I've thought about ways to make stuff like `series_0|input_mode|....` easier, but I haven't really come up with anything good. I kinda wish Galaxy just enforced that the parameter names needed to be unique, and then it would know exactly which param with just the name.....